src/App.tsx

```diff
@@ -1,14 +1,53 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropList } from './components/DropList';
+
+const debounce = (f: (...args: string[]) => void, delay: number) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [findPerson, setFindPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [apllieQuery, setApllieQuery] = useState('');
+
+  const apllyQuery = useCallback(
+    debounce(setApllieQuery, 1000),
```

Comment:
move this value to a constant (outside the component)
const DEBOUNCE_DELAY = 1000;